### DISCUSSION TITLE
Add termux as 3rd party library

### DISCUSSION THREAD

**Discussion by jhay06** [termux/termux-app#3699] [Q&A] [answered]
is there a way we can add termux bash emulator as library to our android app? .

**Comment by twaik** [comment]
You can start commands via intents. https://github.com/termux/termux-app/wiki/RUN_COMMAND-Intent

**Comment by agnostic-apollo** [comment, marked as answer]
If you want to integrate termux into your own app instead of termux being installed separately, it is possibly, but currently there is no library for it, but is planned. But you can do it.

Check [forking](https://github.com/termux/termux-app#forking). You will have to compile bootstrap for your app package, possibly without apt/dpkg and other unneeded packages. On the app side, check `TermuxActivity` which holds the terminal and starts the `TermuxService`, which holds the `TermuxSession`.
